influxdb3_write/src/write_buffer/flusher.rs

```diff
@@ -0,0 +1,245 @@
+//! Buffers writes and flushes them to the configured wal
+
+use crate::write_buffer::buffer_segment::{BufferedWrite, DatabaseWrite, WriteBatch, WriteSummary};
+use crate::write_buffer::{Error, SegmentState, TableBatch};
+use crate::{wal, SequenceNumber, WalOp};
+use data_types::NamespaceName;
+use observability_deps::tracing::debug;
+use parking_lot::{Mutex, RwLock};
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::{mpsc, oneshot, watch};
+
+// Duration to buffer writes before flushing them to the wal
+const BUFFER_FLUSH_INTERVAL: Duration = Duration::from_millis(10);
+// The maximum number of buffered writes that can be queued up before backpressure is applied
+const BUFFER_CHANNEL_LIMIT: usize = 10_000;
+
+// buffered writes should only fail if the underlying WAL throws an error. They are validated before they
+// are buffered. If there is an error, it'll be here
+#[derive(Debug, Clone)]
+pub enum BufferedWriteResult {
+    Success(WriteSummary),
+    Error(String),
+}
+
+#[derive(Debug)]
+pub struct WriteBufferFlusher {
+    join_handle: Mutex<Option<tokio::task::JoinHandle<()>>>,
+    wal_io_handle: Mutex<Option<std::thread::JoinHandle<()>>>,
+    #[allow(dead_code)]
+    shutdown_tx: watch::Sender<()>,
+    buffer_tx: mpsc::Sender<BufferedWrite>,
+}
+
+impl WriteBufferFlusher {
+    pub fn new(segment_state: Arc<RwLock<SegmentState>>) -> Self {
+        let (shutdown_tx, shutdown_rx) = watch::channel(());
+        let (buffer_tx, buffer_rx) = mpsc::channel(BUFFER_CHANNEL_LIMIT);
+        let (io_flush_tx, io_flush_rx) = std::sync::mpsc::channel();
```

Comment:
You can alias the import with `use std::sync::mpsc as std_mpsc` so that you can just do `std_mpsc::channel()` and keeping things a bit consistent and less verbose looking.